.github/workflows/unittests.yaml

```diff
@@ -15,19 +15,25 @@ jobs:
         os: [ubuntu-latest, macos-latest, windows-latest]
         python-version: ["3.7", "3.8", "3.9", "3.10", "3.11.0-rc.1"]
         exclude:
-          - os: macos-latest
-            python-version: "3.10"
           - os: macos-latest
             python-version: "3.11.0-rc.1"
           - os: windows-latest
             python-version: "3.11.0-rc.1"
+        include:
+          - os: macos-latest
+            python-version: "3.10"
```

Comment:
The only reason that this workaround _seems_ specific to Python 3.10 is that compiling `lxml` against the system-provided `libxml2` produces a resulting package that -- as best as I can tell so far -- doesn't correctly handle (three-or-more-character?) multi-byte Unicode characters in UTF-8 streams during `fromstring` operations.

The same problem does occur on Python 3.9 if we use `pip install --no-binary=lxml`.  That hasn't occurred in practice during our unit testing so far, because a binary wheel is available for `lxml==4.9.1` (latest) in Python 3.9, so `pip` opts to use that.

So ideally on MacOS (`darwin`) platforms, we would want to install from the latest statically-linked `lxml` wheel from PyPi.

We can do that easily during GitHub Actions workflows with a platform-specific matrix variable -- but perhaps we could achieve the same thing in the Python packaging so that all users would benefit?

I've no particular opinion about what to do for other operating systems.

Reply:
...or maybe we should just use `--prefer-binary` for all `pip` install operations during GHA and be done with it.

We don't directly depend upon `lxml`, it's a transitive dependency, so trying to manage that ourselves could be tricky.

Reply:
> ...or maybe we should just use `--prefer-binary` for all `pip` install operations during GHA and be done with it.

That seems risky since we have a decent number of dependencies and it mean we could fall behind on fixes/improvements.  And sadly `--prefer-binary` doesn't seem to support named packages (format control) as an argument, so we can't easily limit it to `lxml` (AFAICT?).

This workaround during unit testing seems light-touch so far - and perhaps adding a note about installation in MacOS environments to the readme.

Reply:
Yep, let's `pip install --binary-only=lxml lxml` on MacOS platforms -- essentially the same as the workaround.

Reply:
Yep, let's `pip install --binary-only=lxml lxml` on MacOS platforms -- essentially the same as the workaround.